rholang/src/test/scala/coop/rchain/rholang/interpreter/accounting/CostAccountingSpec.scala

```diff
@@ -145,25 +157,16 @@ class CostAccountingSpec extends FlatSpec with Matchers {
   }
 
   it should "stop the evaluation of all execution branches when one of them runs out of phlo with a more sophisiticated contract" ignore {
-    val initialPhlo = 150L
-    val contract    = """new loop in {
-                             contract loop(@n) = {
-                               match n {
-                                 0 => Nil
-                                 _ => loop!(n-1)
-                               }
-                             } |
-                             loop!(10)
-                           }""".stripMargin
-
-    val (result, costLog) = evaluateWithCostLog(initialPhlo, contract)
-    result shouldBe Left(OutOfPhlogistonsError)
-    val costs = costLog.map(_.value).toList
-    // The sum of all costs but last needs to be <= initialPhlo, otherwise
-    // the last cost should have not been logged
-    costs.init.sum.toLong should be <= (initialPhlo)
-    // The sum of ALL costs needs to be > initialPhlo, otherwise an error
-    // should not have been reported
-    costs.sum.toLong should be > (initialPhlo)
+    check(forAllNoShrink(Gen.choose(1L, shortslow.cost - 1)) { initialPhlo =>
```

Comment:
So, we should generalize this test to take an arbitrary contract rather than just `shortslow`. For `n` processes running in parallel, ensure that  all terminate when one of them runs out of phlo.

Reply:
Yes, the end goal here is to use the par generator (when it's ready)

Reply:
as discussed offline, we don't need to wait for its readiness. We can generate n processes by choosing (with repetitions) from the k predefined processes.

Reply:
I think the tests we have so far are insufficient for testing actual parallel execution and short-circuiting in such an environment. I also think we won't be able to test that reliably without controlling the par shape manually. Change my mind.

Reply:
You're commenting an outdated version so I'm not sure whether you've seen the latest changes. The 3 contracts used in the latest version have some degree of parallelism. You claim that this is not enough for actual testing. What is enough?

The tests are going to be executed multiple times for each contract (with different max phlo). This means the contracts shouldn't be too big, otherwise the tests will take ages to execute. The fact that each test is going to be executed many times gives a good chance of finding concurrency related problems